app/helpers/utm_link_helper.rb

```diff
@@ -0,0 +1,28 @@
+module UtmLinkHelper
+  UTM_PARAMS = {
+    utm_source: 'apply-for-teacher-training.service.gov.uk',
+    utm_medium: 'referral',
+  }.freeze
+
+  def govuk_link_to_with_utm_params(link_text, link_url, utm_campaign, utm_content = nil, **extra_options)
+    url = construct_url(link_url, utm_campaign, utm_content)
+    govuk_link_to link_text.to_s, HostingEnvironment.production? ? url.to_s : link_url, extra_options
+  end
+
+  def email_link_with_utm_params(link_url, utm_campaign, utm_content)
+    url = construct_url(link_url, utm_campaign, utm_content)
+    HostingEnvironment.production? ? url.to_s : link_url
+  end
+
+  def utm_campaign(params)
+    params.slice(:controller, :action).values.join('-')
+  end
+
+private
+
+  def construct_url(link_url, utm_campaign, utm_content)
```

Comment:
if you put a guard clause in here:
```
return link_url unless HostingEnvironment.production?
```
you could avoid the ternary condition on lines 9 and 14. i think this would be a bit clearer.

Reply:
much cleaner, thanks